pkg/signal/trap.go

```diff
@@ -14,41 +15,56 @@ import (
 // (and the Docker engine in particular).
 //
 // * If SIGINT or SIGTERM are received, `cleanup` is called, then the process is terminated.
-// * If SIGINT or SIGTERM are repeated 3 times before cleanup is complete, then cleanup is
-// skipped and the process terminated directly.
-// * If "DEBUG" is set in the environment, SIGQUIT causes an exit without cleanup.
+// * If SIGINT or SIGTERM are received 3 times before cleanup is complete, then cleanup is
+//   skipped and the process is terminated immediately (allows force quit of stuck daemon)
+// * A SIGQUIT always causes an exit without cleanup, with a goroutine dump preceding exit.
+// * A SIGUSR1 does not exit the daemon, but simply dumps goroutine and any other pertinent
+//   debug information
 //
 func Trap(cleanup func()) {
+	// separate handler for SIGUSR1 (due to no Windows support for SIGUSR1)
+	// handles goroutine dumping
+	setupSigusr1Trap()
```

Comment:
https://golang.org/pkg/os/signal/#Notify shows indeed that it doesn't matter, so it's okay